### DISCUSSION TITLE
Resolve Faucet Proposal confliction

### DISCUSSION THREAD

**Discussion by manishram** [thenewboston-developers/Project-Proposals#145] [General]
We need to come to conclusions for Faucet Proposal.

@itsnikhil @mrbusysky

**Comment by itsnikhil** [comment]
@mrbusysky if you want to provide educational platform to teach working of thenewboston then I am all for it but I don't see any reason why you are seeking funding for a project which we have already worked. If your "educational site" really needs to have a service like Faucet then you can use ours. You can even use our code but dont ask funding for the same

**Reply by mrbusysky** [reply]
The main script would be a blank script anyone can use and modify with out having to remove features. This would allow any website to start from a blank clean slate like most chains have.

Your script has a requirement that they must tweet when not every site will need or want this feature.

**Reply by itsnikhil** [reply]
Tweet is a restriction to prevent abuse and marketing. If you have issues with that, you can suggest improvement via github issue. Also, trust me we have made it and it is not as simple as a script!

**Reply by mrbusysky** [reply]
The point is not every website needs this feature so they will have to go through the trouble of stripping it for their own purposes. This is a script your what you find acceptable as valid. Not every service in network is going to come to that same conclusion.

Providing a base script anyone can work from and adding features on it that allow for mass adoption quicker is ideal. So I won't be canceling.

**Reply by itsnikhil** [reply]
If everyone starts hosting their own banks and validators, it completely neglects the purpose of having a faucet

**Reply by mrbusysky** [reply]
Not in the service economy of the chain

**Reply by itsnikhil** [reply]
@mrbusysky whatever you are saying right now is completely different from what you have written in project description. And I am not able to interpret whatever you are saying right now

**Reply by mrbusysky** [reply]
Also, the new system is different. Best to wait to talk about this until bucky posts the new document today

**Reply by mrbusysky** [reply]
There will the base script which will be separated from the main project the main project will also be made ontop of the base script

**Reply by itsnikhil** [reply]
Base script is like a docker image?

**Reply by itsnikhil** [reply]
If the system changes, we will update our Faucet to the latest version. So thenewboston don't have to fund other projects to create a new solution from scratch.

**Reply by mrbusysky** [reply]
Yeah it would be like a docker image.. And I have no clue what the second link is for. I was sent this at 5am telling me needed to solve this issue while I'm in bed typing on a phone.

**Reply by mrbusysky** [reply]
It doesn't change the fact that your project isn't useful to people looking to start from scratch. They must remove everything you added

**Reply by mrbusysky** [reply]
This is how the base script needs to be http://salmen.website/Faucet/

**Reply by mrbusysky** [reply]
https://github.com/Salmen2/Simple-Faucet-Script

**Reply by itsnikhil** [reply]
It would be better to follow up on this tomorrow as Bucky also shares the "base script".

By that link I meant you cannot communicate your point properly to me. If base script is docker image like that of Bank. You should know running bank node doesn't yield you coins which you can distribute. Whatever you mean by sharing a scipt went on top of my head.

Let me explain you how our system works, and hope you can explain your system with such clarity

Faucet is to distribute testnet coins so that people can use test coins instead of burning real coins.
We have our own set of servers running aka testnet so that we have access to all the fake coins and others don't have to run their own servers to learn test application. Regarding distribution as I have already explained it is what Rinkeby - https://faucet.rinkeby.io/ faucet also do.

According to your project description you want to implement the exact same thing against which this whole conversation is going on.

**Reply by itsnikhil** [reply]
> It doesn't change the fact that your project isn't useful to people looking to start from scratch

@mrbusysky I ask you a simple question. Have you tried our Faucet?
What you are saying proves that you don't understand how our Faucet works. I have told you before to go through the FAQ on the page. People just need to change to our testnet bank. Our testnet works with existing Wallets

**Reply by itsnikhil** [reply]
If people were to host their own Faucet and setup their own test network then the code is already there. They should already know to change root account file to their own address and they will have access to all the coins in treasury. They don't need a Faucet at that point, they can use Account manager or any other wallet to exchange coins. Faucet is there so that people don't need to setup their own testnet.

**Comment by itsnikhil** [comment]
@manishram  @mrbusysky you can go ahead with your project! I had wrong interpretation of your project
